light-client/cli/src/bin/main.rs

```diff
@@ -2,27 +2,61 @@
 
 use clap::{Parser, Subcommand};
 use grpcio::{ChannelBuilder, EnvBuilder};
+use mc_api::blockchain::ArchiveBlocks;
+use mc_blockchain_types::BlockIndex;
 use mc_common::{
-    logger::{create_app_logger, o},
+    logger::{create_app_logger, log, o, Logger},
     ResponderId,
 };
 use mc_consensus_api::{
     consensus_client_grpc::ConsensusClientApiClient, consensus_common_grpc::BlockchainApiClient,
 };
 use mc_consensus_scp_types::QuorumSet;
+use mc_ledger_sync::ReqwestTransactionsFetcher;
 use mc_light_client_verifier::{
-    HexKeyNodeID, LightClientVerifierConfig, TrustedValidatorSetConfig,
+    HexKeyNodeID, LightClientVerifier, LightClientVerifierConfig, TrustedValidatorSetConfig,
 };
 use mc_util_grpc::ConnectionUriGrpcioChannel;
 use mc_util_uri::ConsensusClientUri;
-use std::{str::FromStr, sync::Arc};
+use protobuf::Message;
+use std::{fs, path::PathBuf, str::FromStr, sync::Arc};
 
 #[derive(Subcommand)]
 pub enum Commands {
+    /// Generate a light client verifier config from a list of nodes.
+    /// This does not include any historical data.
     GenerateConfig {
+        /// Node URIs to use for generating the config.
         #[clap(long = "node", use_value_delimiter = true, env = "MC_NODES")]
         nodes: Vec<ConsensusClientUri>,
     },
+
+    /// Fetch one or more `[ArchiveBlock]`s from a list of tx source urls and
+    /// store them in a Protobuf file.
+    FetchArchiveBlocks {
+        /// URLs to use for fetching blocks.
+        ///
+        /// For example: https://ledger.mobilecoinww.com/node1.prod.mobilecoinww.com
+        #[clap(
+            long = "tx-source-url",
+            use_value_delimiter = true,
+            env = "MC_TX_SOURCE_URL"
+        )]
+        tx_source_urls: Vec<String>,
+
+        /// Block index we are interested in.
+        #[clap(long, env = "MC_BLOCK_INDEX")]
+        block_index: BlockIndex,
+
+        /// File to write the fetched ArchiveBlocks protobuf to.
+        #[clap(long, env = "MC_OUT_FILE")]
+        out_file: PathBuf,
+
```

Comment:
:upside_down_face: This behavior of requiring an output file seems inconsistent with the `GenerateConfig` command.
One writes JSON to stdout, another requires a file.

Has something like https://docs.rs/clio/latest/clio/ been considered to make these CLI's a bit more consistent?

Reply:
No, but that looks like a nice crate so I'll switch to that.